pkg/db/queries.sql

```diff
@@ -75,3 +75,56 @@ FROM
 GROUP BY
 	originator_node_id;
 
+-- name: GetAddressLogs :many
+SELECT
+	a.address,
+	encode(a.inbox_id, 'hex') AS inbox_id,
+	a.association_sequence_id
+FROM
+	address_log a
+	INNER JOIN (
+		SELECT
+			address,
+			MAX(association_sequence_id) AS max_association_sequence_id
+		FROM
+			address_log
+		WHERE
+			address = ANY (@addresses::TEXT[])
+			AND revocation_sequence_id IS NULL
+		GROUP BY
+			address) b ON a.address = b.address
+	AND a.association_sequence_id = b.max_association_sequence_id;
+
+-- name: InsertAddressLog :one
+INSERT INTO address_log(address, inbox_id, association_sequence_id, revocation_sequence_id)
+	VALUES (@address, decode(@inbox_id, 'hex'), @association_sequence_id, @revocation_sequence_id)
+RETURNING
+	*;
+
+-- name: RevokeAddressFromLog :exec
+UPDATE
+	address_log
+SET
+	revocation_sequence_id = @revocation_sequence_id
+WHERE (address, inbox_id, association_sequence_id) =(
+	SELECT
+		address,
+		inbox_id,
+		MAX(association_sequence_id)
+	FROM
+		address_log AS a
+	WHERE
+		a.address = @address
+		AND a.inbox_id = decode(@inbox_id, 'hex')
+	GROUP BY
+		address,
+		inbox_id);
```

Comment:
Do we need this sub-query, given `(address, inbox_id)` is a primary key that uniquely identifies the row? The PK is a small simplifying change compared to the old server

Reply:
Not sure I follow the suggestion. IIRC we need the subquery because an address can be associated with multiple inbox IDs, and we need to find the one with the highest `association_sequence_id`?

Maybe you can sketch out what you have in mind as an alternate.

Reply:
I totally misread what this was doing, my bad!